.github/workflows/wheels.yml

```diff
@@ -2,7 +2,7 @@ name: cibuildwheel
 # https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency
 # https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types#pullrequestevent
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.number }}-${{ github.event.type }}
+  group: ${{ github.ref }}-${{ github.event.type }}
```

Comment:
By getting rid of `workflow` aren't you making `wheels.yml` workflow and `main.yml` workflow in the same group?

In another project I've successfully been using:

```
group: ${{ github.workflow }}-${{ github.event.number || github.event.ref }}-${{ github.event.type }}
```